python/src/scipp/plot/plot_3d.py

```diff
@@ -82,292 +87,546 @@ def __init__(self,
+        self.outline, self.axticks = self.create_outline()
+
+        # Save the size of the outline box for later
+        self.box_size = np.diff(list(self.xminmax.values()), axis=1).ravel()
+
+        # Define camera: look at the centre of mass of the points
+        camera_lookat = self.center_of_mass
+        camera_pos = np.array(self.center_of_mass) + 1.2 * self.box_size
+        self.camera = p3.PerspectiveCamera(position=list(camera_pos),
+                                           aspect=config.plot.width /
+                                           config.plot.height)
+
+        # Add red/green/blue axes helper
+        self.axes_3d = p3.AxesHelper(10.0 * np.linalg.norm(camera_pos))
+
+        # Create the pythreejs scene
+        self.scene = p3.Scene(children=[
+            self.camera, self.axes_3d, self.points, self.outline, self.axticks
+        ],
+                              background=background)
+
+        # Add camera controller
+        self.controller = p3.OrbitControls(controlling=self.camera,
+                                           target=camera_lookat)
+        self.camera.lookAt(camera_lookat)
+
+        # Render the scene into a widget
+        self.renderer = p3.Renderer(camera=self.camera,
+                                    scene=self.scene,
+                                    controls=[self.controller],
+                                    width=config.plot.width,
+                                    height=config.plot.height)
+
+        # Opacity slider: top value controls opacity if no cut surface is
+        # active. If a cut curface is present, the upper slider is the opacity
+        # of the slice, while the lower slider value is the opacity of the
+        # data not in the cut surface.
+        self.opacity_slider = widgets.FloatRangeSlider(min=0.0,
+                                                       max=1.0,
+                                                       value=[0.1, 1],
+                                                       step=0.01,
+                                                       description="Opacity",
+                                                       continuous_update=True)
+        self.opacity_slider.observe(self.update_opacity, names="value")
+        self.opacity_checkbox = widgets.Checkbox(
+            value=self.opacity_slider.continuous_update,
+            tooltip="Continuous update",
+            indent=False,
+            layout={"width": "initial"})
+        self.opacity_checkbox_link = widgets.jslink(
+            (self.opacity_checkbox, 'value'),
+            (self.opacity_slider, 'continuous_update'))
+
+        # Add buttons to provide a choice of different cut surfaces:
+        # - Cartesian X, Y, Z
+        # - Cylindrical X, Y, Z (cylinder major axis)
+        # - Sperical R
+        # - Value-based iso-surface
+        self.cut_surface_buttons = widgets.ToggleButtons(
+            options=[('X ', 0), ('Y ', 1), ('Z ', 2), ('R ', 6), (' X ', 3),
```

Comment:
Suggestion - Could you represent options better with something like Enum
```python
from enum import Enum
class CutOption(Enum):
    X = 0
    ....
```
Then further down when you are interacting with chosen options should be clearer what is going on

Reply:
Good suggestion. I started with an `Enum`, as you suggested, but was having to do things like `self.cut_surface_buttons.value.value`, which was also a little confusing.
In the end, I just used a `dict`.